tests/chainer_tests/functions_tests/rnn_tests/test_function_tree_lstm.py

```diff
@@ -0,0 +1,106 @@
+import six
+
+import numpy
+
+from chainer import functions
+from chainer import testing
+from chainer.testing import backend
+
+
+def _sigmoid(x):
+    half = x.dtype.type(0.5)
+    return numpy.tanh(x * half) * half + half
+
+
+def _shaped_random_array(shape, dtype):
+    return numpy.random.uniform(-1, 1, shape).astype(dtype)
+
+
+@testing.parameterize(*testing.product_dict(
+    [
+        {'c_dim': 2, 'num_c': 1, 'batch_size': 3},
+        {'c_dim': 0, 'num_c': 4, 'batch_size': 5},
+        {'c_dim': 12, 'num_c': 1, 'batch_size': 0},
+    ], [
+        {'dtype': numpy.float16},
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float64},
+    ]
+))
+@testing.fix_random()
+@backend.inject_backend_tests(
+    None,
+    # ChainerX tests
+    testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+    # CPU tests
+    + testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product([
+        [{'use_cuda': True}],
+
+        # Without cuDNN
+        testing.product({
+            'use_cudnn': ['never'],
+        })
+        # With cuDNN
+        + testing.product({
+            'use_cudnn': ['always'],
+            'cudnn_deterministic': [True, False],
+            'autotune': [True, False],
+        })]))
+class TestTreeLSTM(testing.FunctionTestCase):
+
+    dodge_nondifferentiable = True
+
+    def setUp(self):
+        self.check_forward_options.update({
+            'rtol': 1e-2, 'atol': 1e-2})
+        self.check_backward_options.update({
+            'rtol': 1e-2, 'atol': 1e-2})
+        self.check_double_backward_options.update({
+            'rtol': 5e-3, 'atol': 5e-2})
+        self.skip_double_backward_test = True
```

Comment:
Could you explain why double backward tests should be skipped?

Reply:
I am getting the following error while running the backward test:
`RuntimeError: cannot twice-differentiate an old style Function "TreeLSTM"`

Reply:
I see. Could you leave a comment about it in the description of this PR?

Reply:
Done.